core/test-src/org/pentaho/di/core/vfs/KettleVFSTest.java

```diff
@@ -0,0 +1,39 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2015 by Pentaho : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.core.vfs;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class KettleVFSTest {
+
+  @Test
+  public void checkFriendlyURI() {
```

Comment:
Please add a few more asserts for other types of URIs such as:
*  a relative path: "./subdir/file.txt"
*  a file: scheme: "file:///tmp/test.txt"
*  hdfs schemes with and without passwords
*  ftp, sftp, ftps - with and without password
*  zip, gzip, tar
*  a complex wrapped scheme: tar:gz: hdfs://user:pass@somehost:8080/downloads/some_file.tar.gz!/mytar.tar!/path/in/tar/README.txt

Note, I'm not sure whether your test is ever actually hitting the VFS code.  It might be only following the exception path since these tests were not actually valid files.  Try testing some actual files that exist in the test resources directory using the fileObject signature.